### DISCUSSION TITLE
the key of hash type in redis

### DISCUSSION THREAD

**Discussion by touchsky2018** [Budibase/budibase#7558] [Help] [answered]
how to query redis with the key of hash type

**Comment by melohagan** [comment, marked as answer]
Hi @touchsky2018 

I've added some docs for Redis here: https://docs.budibase.com/docs/redis

You will see how a hash type can be set in the Redis Command screenshot: https://docs.budibase.com/docs/redis#redis-command

You could then get the *firstName* with the following command: `hget profile firstName`

More in the official Redis docs here: https://redis.io/docs/data-types/hashes/

**Reply by rito** [reply]
when i create an item in redis i can only create type String. How is it possible to create more flexible items like a Hash with multiple options? Just the String type is not viable for my application.

**Reply by melohagan** [reply]
Hey @rito It's explained in the docs, use the **Redis Command** function:
https://docs.budibase.com/docs/redis#redis-command
